Runtime/Runner/Scenes/LanguageSelectorController.cs

```diff
@@ -0,0 +1,240 @@
+    }
+
+    public void SetLanguageFromTitle(string title) {
+        string  lang = GetLanguageFromTitle(title);
+        if (lang == null) {
+            Debug.LogError("Language not found. (" + title + ")");
+        } else {
+            Language = title;
+        }
+    }
+
+    //Selects a language by flag button in Title scene
+    public void fillDictionaryAndRunLoginScene()
+    {
+        SetUpJSONFiles(false);
+        FillDictionary(false);
+        SimvaPlugin.Instance.RunScene("Simva.Login");
+    }
+
+      //Selects a language by flag button in Title scene
+    public void SetLanguageCode(string code)
+    {
+        Language = code;
+    }
+
+    public void SetActive(bool active)
+    {
+        this.gameObject.SetActive(active);
+    }
+
+    public string GetCurrentLanguage() { return Language; }
+
+    public void RefreshLanguageList()
+    {
+        // Clear existing children
+        if(languageGridLayout) {
+            foreach (Transform child in languageGridLayout.transform)
+            {
+                Destroy(child.gameObject);
+            }
+        }
+
+        // Spawn one UI item per selected language
+        foreach (string lang in SimvaPlugin.Instance.SelectedLanguages)
+        {
+            GameObject item = Instantiate(languageItemPrefab, languageGridLayout.transform);
+            foreach(string languageCode in languages.Keys) {
+                if(languages[languageCode] == lang) {
+                    item.name = languageCode;
+                    item.SetActive(true);
+                    continue;
+                }
+            }
+        }
+    }
+
+    void GetLanguages()
+    {
+        languages = new Dictionary<string, string>();
+
+        TextAsset[] filler = Resources.LoadAll<TextAsset>("Localization");
+        foreach (TextAsset obj in filler) {
+            if (obj.name == "lang") {
+                JObject jObject = JObject.Parse(obj.text);
+                var code="";
+                var name="";
+                var custom=false;
+                foreach (var entry in jObject) {
+                    if(entry.Key=="code") {
+                        code=(string)entry.Value;
+                    }
+                    if(entry.Key=="displayName") {
+                        name=(string)entry.Value;
+                    }
+                    if(entry.Key=="custom") {
+                        custom=(bool)entry.Value;
+                    }
+                }
+                var modifName=name + " [" + code + "]";
+                if(!languages.ContainsKey(code)) {
+                    languages.Add(code, modifName);
+                } else {
+                    //if(custom) {
+                    //    languages.Add(code, modifName);
+                    //}
+                }
+            }
+        }
+        Debug.Log("Languages : " + languages.Count);
+    }
+
+    //Creates an initializes Json Files array to be used by myDictionary
+    void SetUpJSONFiles(bool useDefault)
```

Comment:
This method should return the json at the end and should receive the language by parameter.
They way it is implemented right now is obscure and changes different internal variables which makes the code more difficult to understand.
I suggest changing the signature to "List\<TextAsset\> LoadLanguageJSON(string language)" and passing the language by default in awake and passing Language in FillDictionaryAndRunLoginScene.